katdal/datasources.py

```diff
@@ -643,6 +644,16 @@ def from_url(cls, url, chunk_store='auto', upgrade_flags=True, **kwargs):
                 telstate = katsdptelstate.TelescopeState(url_parts.netloc, db)
             except katsdptelstate.ConnectionError as e:
                 raise DataSourceNotFound(str(e))
+        elif url_parts.scheme in {'http', 'https'}:
+            # Open endpoint as an S3 object store (with auth info in kwargs)
+            endpoint = urllib.parse.urlunparse(url_parts[:2] + 4 * ('',))
+            rdb_store = S3ChunkStore.from_url(endpoint, **kwargs)
+            rdb_path = urllib.parse.urlunparse(url_parts[:3] + 3 * ('',))
```

Comment:
I think we should respect the nesting level of the request path, so that if the URL given is `https://foo.bar/level1/level2/123456789/123456789_sdp_l0.rdb', then we should construct the chunk store for `https://foo.bar/level1/level2`. We probably won't need it ourselves, but conceivable one might have a reverse proxy in front of a chunk store that maps it to a subpath.

Reply:
Done.